src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -54,21 +54,37 @@ internal sealed class PdbSourceDocumentMetadataAsSourceFileProvider : IMetadataA
             if (signaturesOnly)
                 return null;
 
+            using var _ = TelemetryHelper.Start(cancellationToken);
+
+            var assemblyName = symbol.ContainingAssembly.Identity.Name;
+
+            if (_logger is not null)
+            {
+                // We block to clear the log from the previous operation, so things don't get confusing
+                // if the log messages are delayed
+                await _logger.ClearAsync().ConfigureAwait(false);
+            }
+
+            _logger?.Log(FeaturesResources.Navigating_to_symbol_0_from_1, symbol, assemblyName);
+
             // If this is a reference assembly then we won't have the right information available, so bail out
             // TODO: find the implementation assembly for the reference assembly, and keep going: https://github.com/dotnet/roslyn/issues/55834
             var isReferenceAssembly = symbol.ContainingAssembly.GetAttributes().Any(attribute => attribute.AttributeClass?.Name == nameof(ReferenceAssemblyAttribute)
                 && attribute.AttributeClass.ToNameDisplayString() == typeof(ReferenceAssemblyAttribute).FullName);
             if (isReferenceAssembly)
+            {
+                _logger?.Log(FeaturesResources.Source_is_a_reference_assembly);
                 return null;
+            }
 
             var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
 
+            // The purpose of the logging is to help library authors, so we don't log things like this where something
```

Comment:
This comment seems weirdly placed. Is this for line 87 below?

Reply:
No, this is meant to be here to justify why we're _not_ logging if we exit early here, but we did log when exiting early in the reference assembly check above. I'll reword it a bit.